CRM/Contribute/Form/Task/PDF.php

```diff
@@ -182,6 +182,15 @@ public function postProcess() {
       $input['trxn_date'] = isset($contribution->trxn_date) ? $contribution->trxn_date : NULL;
       $input['receipt_update'] = $params['receipt_update'];
       $input['contribution_status_id'] = $contribution->contribution_status_id;
+      $payment_processor_id = NULL;
+      $processorDAO = CRM_Core_DAO::executeQuery("SELECT payment_processor_id
```

Comment:
If you did "limit 1 " you could use singleValueQuery & save a few lines of code...